cli/assets.go

```diff
@@ -256,3 +270,318 @@ func deleteAssetByIDCommand() *cobra.Command {
+			report = append(report, []string{"NAME", "COUNT"})
+			index := 1
+			for _, i := range res.GetData() {
+				report = append(report, []string{term.Bluef(i.Name), fmt.Sprintf("%v", i.Count)})
+				index++
+			}
+			printer.Table(os.Stdout, report)
+
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&types, "types", "t", "", "filter by types")
+	cmd.Flags().StringVarP(&services, "services", "s", "", "filter by services")
+	cmd.Flags().StringVarP(&data, "data", "d", "", "filter by field in asset.data")
+	cmd.Flags().StringVar(&q, "query", "", "filter by specific query")
+	cmd.Flags().StringVar(&q_fields, "query_fields", "", "filter by query field")
+
+	return cmd
+}
+
+func makeGetAllTypesRequest(types, services, data, q, q_fields string) *compassv1beta1.GetAllTypesRequest {
+	newReq := &compassv1beta1.GetAllTypesRequest{}
+	if types != "" {
+		newReq.Types = types
+	}
+	if services != "" {
+		newReq.Services = services
+	}
+	if q != "" {
+		newReq.Q = q
+	}
+	if q_fields != "" {
+		newReq.QFields = q_fields
+	}
+	if data != "" {
+		newReq.Data = makeMapFromString(data)
+	}
+	return newReq
+}
+
+func listAssetStargazerCommand() *cobra.Command {
+	var size, page uint32
+	cmd := &cobra.Command{
+		Use:   "stargazers <id>",
+		Short: "list all stargazers for a given asset id",
+		Example: heredoc.Doc(`
+			$ compass asset stargazers <id>
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			assetID := args[0]
+			ctx := client.SetMetadata(cmd.Context())
+			res, err := clnt.GetAssetStargazers(ctx, &compassv1beta1.GetAssetStargazersRequest{
+				Id:     assetID,
+				Size:   size,
+				Offset: page,
+			})
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+			fmt.Println(term.Bluef(prettyPrint(res.GetData())))
+			return nil
+		},
+	}
+	cmd.Flags().Uint32Var(&size, "size", 20, "Size of each page")
+	cmd.Flags().Uint32Var(&page, "page", 0, "Number of pages")
+
+	return cmd
+}
+
+func starAssetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "star <id>",
+		Short: "star an asset by id for current user",
+		Example: heredoc.Doc(`
+			$ compass asset star <id>
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			assetID := args[0]
+			ctx := client.SetMetadata(cmd.Context())
+			_, err = clnt.StarAsset(ctx, &compassv1beta1.StarAssetRequest{
+				AssetId: assetID,
+			})
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			fmt.Println(term.Bluef("Asset %v starred successfully", assetID))
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func unstarAssetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unstar <id>",
```

Comment:
As most of the added commands expect a positional argument, I would advice we can use a validation check as
```go
Args: cobra.MinimumNArgs(1),
```